src/CachePlugin.php

```diff
@@ -293,16 +293,16 @@ private function getETag(CacheItemInterface $cacheItem)
             return;
         }
 
-        if (is_array($data['etag'])) {
-            foreach ($data['etag'] as $etag) {
-                if (!empty($etag)) {
-                    return $etag;
-                }
-            }
+        if (!is_array($data['etag'])) {
+            return $data['etag'];
+        }
 
-            return;
+        foreach ($data['etag'] as $etag) {
+            if (!empty($etag)) {
+                return $etag;
+            }
         }
 
-        return $data['etag'];
+        return;
```

Comment:
Yes, it would be the same result if we remove this. But that is technically wrong. Because `void !== null`.